polaris-react/rollup.config.js

```diff
@@ -0,0 +1,100 @@
+const path = require('path');
+
+const {babel} = require('@rollup/plugin-babel');
+const commonjs = require('@rollup/plugin-commonjs');
+const {nodeResolve} = require('@rollup/plugin-node-resolve');
+const externals = require('rollup-plugin-node-externals');
+const replace = require('@rollup/plugin-replace');
+const image = require('@rollup/plugin-image');
+const json = require('@rollup/plugin-json');
+
+const {styles} = require('./config/rollup/plugin-styles');
+const {generateScopedName} = require('./config/rollup/namespaced-classname');
+const postcssPlugins = require('./config/postcss-plugins');
+const pkg = require('./package.json');
+
+function generateConfig({output, targets, stylesConfig}) {
+  return {
+    input: './src/index.ts',
+    plugins: [
+      externals({deps: true, packagePath: './package.json'}),
+      nodeResolve({extensions: ['.js', '.jsx', '.ts', '.tsx']}),
+      commonjs(),
+      babel({
+        rootMode: 'upward',
+        extensions: ['.js', '.jsx', '.ts', '.tsx'],
+        exclude: 'node_modules/**',
+        babelHelpers: 'bundled',
+        // Options that may be present on the `babelConfig` object but
+        // we want to override
+        envName: 'production',
+        // @ts-expect-error targets is a valid babel option but @types/babel__core doesn't know that yet
+        targets,
+      }),
+      replace({
+        '{{POLARIS_VERSION}}': pkg.version,
+        delimiters: ['', ''],
+        preventAssignment: true,
+      }),
+      image(),
+      json({
+        compact: true,
+      }),
+      styles(stylesConfig),
+    ],
+    output,
+  };
+}
+
+/**
+ * @type {import('rollup').RollupOptions}
+ */
```

Comment:
Nitpick: you can specify these docblocks on one line

```suggestion
/** @type {import('rollup').RollupOptions} */
```